source/portable/NetworkInterface/DriverSAM/gmac_SAM.c

```diff
@@ -621,6 +621,9 @@ uint32_t gmac_dev_read( gmac_device_t * p_gmac_dev,
         return GMAC_RX_NO_DATA;
     }
 
+    /* Return the number of bytes received. */
+    *p_rcv_size = bytesLeft;
```

Comment:
```suggestion
    *p_rcv_size = min( bytesLeft + 2, ( int32_t ) ul_frame_size );
```

Reply:
@ActoryOu and @htibosch this should be the fix. Am I correct?

Reply:
Hi @AniruddhaKanhere,
It can't be `bytesLeft + 2` because the stack compares packet length with buffer length.
It's only allowed when buffer length is exactly equal to packet length, which is bytesLeft.
Take #941 as reference.